Formula/libxml2.rb

```diff
@@ -41,7 +41,7 @@ class Libxml2 < Formula
 
   keg_only :provided_by_osx
 
-  depends_on :python => :optional
+  depends_on :python if MacOS.version <= :snow_leopard
```

Comment:
This is an interesting conversation but it is not related to this PR; after this PR, the formula will use system Python.

I still think making Python keg-only will only cause greater confusion; I favor my homebrew-evolution proposal.